### ISSUE TITLE
variant synchronisation delete and add image url, leading to image deleted on CDN

### ISSUE THREAD

**Issue by yanns** [sphereio/sphere-node-product-csv-sync#111]
The synchronisation has deleted and added an image url. The side effect is that the image is deleted in the CDN and is lost.

The events:
```
{
    "_id" : [...]
    "topic" : "products",
    "events" : [ 
        {
            "event" : {
                "type" : "ProductAttributeChanged",
                "productId" : "<product-id>",
                "variantId" : 1,...
            }
        }, 
        {
            "event" : {
                "type" : "ProductAttributeAdded",
                "productId" : "<product-id>",
                "variantId" : 1,...
            }
        }, 
        {
            "event" : {
                "type" : "ProductAttributeAdded",
                "productId" : "<product-id>",
                "variantId" : 1,...
            }
        }, 
        {
            "event" : {
                "type" : "ProductImageRemoved",
                "productId" : "<product-id>",
                "variantId" : 1,
                "imageUrl" : "<image-url>",...
            }
        }, 
        {
            "event" : {
                "type" : "ProductImageAdded",
                "productId" : "<product-id>",
                "variantId" : 1,
                "image" : {
                    "url" : "<image-url>",
                    "dimensions" : {
                        "w" : 0,
                        "h" : 0
                    }
                },...
            }
        }
    ],...
}
```

**Comment by stmeissner** [comment]
@PhilippSpo It happened again that a live project lost images due to this issue.

**Comment by hajoeichler** [comment]
The problem here is that the tool provides a default dimension of x:0 and y:0 when transforming from Csv to JSON. And this results in a difference why the image is deleted and readded.

**Comment by emmenko** [comment]
@hajoeichler right, I thought it was something like that. What should we do then?

**Comment by stmeissner** [comment]
Has this been fixed along with https://github.com/sphereio/sphere-node-sdk/issues/114 ?

**Comment by PhilippSpo** [comment]
This has been fixed in the SDK, but the csv-sync needs to be updated to use the new SDK version yet.

**Comment by PhilippSpo** [comment]
now it is fixed with the v0.8.0 [release](https://github.com/sphereio/sphere-node-product-csv-sync/commit/d61e2c255cb448ba13636569a36c5e9624ef5ec6)

**Comment by yanns** [comment]
:trophy:
